sites/all/modules/reol_prevent_unpublish/reol_prevent_unpublish.module

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Prevent selected node from being unpublished.
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function reol_prevent_unpublish_menu() {
+  $items = [];
+
+  $items['admin/config/ereolen/prevent'] = array(
+    'title' => 'Prevent un-publish',
+    'description' => 'Control which nodes to protected.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('reol_prevent_unpublish_settings_form'),
+    'access arguments' => array('administer content'),
+    'file' => 'includes/reol_prevent_unpublish.admin.inc',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_node_presave().
+ *
+ * Ensure no one can unpublish the nodes in any-way.
+ */
+function reol_prevent_unpublish_node_presave($node) {
+  if (0 === $node->status) {
+    $nids = array_map('trim', explode(',', variable_get('reol_prevent_unpublish_nid', '')));
+    if (in_array($node->nid, $nids)) {
```

Comment:
Consider moving this to a helper function, `_reol_prevent_unpublish_is_protected($nid)`, say, to handle the parsing of the value of `reol_prevent_unpublish_nid` variable in only one place. Currently this code almost duplicates lines 47–48.

Reply:
Function added :-)